oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/HttpSessionOAuth2AuthorizationRequestRepository.java

```diff
@@ -67,7 +70,13 @@ public void saveAuthorizationRequest(OAuth2AuthorizationRequest authorizationReq
 		}
 		String state = authorizationRequest.getState();
 		Assert.hasText(state, "authorizationRequest.state cannot be empty");
-		Map<String, OAuth2AuthorizationRequest> authorizationRequests = this.getAuthorizationRequests(request);
+		Map<String, OAuth2AuthorizationRequest> authorizationRequests;
+		if (this.allowMultipleAuthorizationRequests) {
+			authorizationRequests = this.getAuthorizationRequests(request);
+		}
+		else {
+			authorizationRequests = new HashMap<>();
```

Comment:
I thought of that too. But that would make the serialization data incompatible; if a user switched between `allowMultipleAuthorizationRequests` then they would get lots of warnings logged. Maybe that's a small annoyance, but it still seems to be an issue that would best be avoided.

Reply:
@jgrandja _ping_
I'd really like to get this work completed... can you please provide me the guidance necessary for that to happen?

Thank you again!

Reply:
Apologies for the delay @candrews. I have a lot of things I'm juggling so I'm just rotating through my priorities. I'll get to this tomorrow. Thanks for your patience.

Reply:
I also have a bunch of priorities, all conflicting and all urgent - I guess that's just how it goes :) Thank you for your continued attention to this issue, I greatly appreciate the opportunity to address it with you.

Reply:
Thank you for thinking about passivity. However, I'd like to consider that we are also deprecating and thus removing `setAllowMultipleAuthorizationRequests` at some point. At that time, if there are values persisted as a `Map`, user applications will break when they update. This is not a huge deal since we will remove the deprecated method on a major release, but it is not ideal.

Instead, I'd prefer to read the session attribute and do an `instanceof` check. If it is a Map the code when the look up the proper value in the Map. Otherwise it would use the `OAuth2AuthorizationRequest` found in the session.

When saving, we would always save according to the `allowMultipleAuthorizationRequests` property.

This has the advantage that if users are not leveraging the deprecated `setAllowMultipleAuthorizationRequests`, then they can expect that an update to Spring Security 6 will be unlikely to cause a `ClassCastException` when they update to it. We also don't need any special code in Spring Security 6 to handle this scenario.